frontend/src/components/table/schedule/schedule.component.tsx

```diff
@@ -0,0 +1,90 @@
+import React, { useEffect } from "react";
+import { useDispatch, useSelector } from "react-redux";
+import { ActionModel } from "../../../state/models/action.model";
+import { ApplicationStateModel } from "../../../state/models/application-state.model";
+import { WorkerType } from "../../../state/models/schedule-data/employee-info.model";
+import { ScheduleDataModel } from "../../../state/models/schedule-data/schedule-data.model";
+import { ScheduleDataActionType } from "../../../state/reducers/schedule-data.reducer";
+import { ScheduleRowComponent } from "./schedule-parts/schedule-row.component";
+import { ScheduleActionType } from "./schedule-state.model";
+import "./schedule.component.css";
+import { ChildrenSectionComponent } from "./sections/children-section/children-section.components";
+import { DateSectionComponent } from "./sections/date-section/date-section.component";
+import { ShiftsSectionComponent } from "./sections/shifts-section/shifts-section";
+import { useScheduleState } from "./use-schedule-state";
+
+export function ScheduleComponent() {
+  const scheduleModel = useSelector((state: ApplicationStateModel) => state.scheduleData);
+  const dispatchGlobalState = useDispatch();
+  const [scheduleState, dispatchScheduleState, setNewSchedule, scheduleLogic] = useScheduleState();
+
+  useEffect(() => {
+    if (scheduleState.isScheduleModified && scheduleLogic) {
+      dispatchGlobalState({
+        type: ScheduleDataActionType.UPDATE,
+        payload: scheduleLogic.getScheduleDataModel(),
+      } as ActionModel<ScheduleDataModel>);
+    }
+  }, [scheduleState, dispatchGlobalState, scheduleLogic]);
```

Comment:
Ogólnie ten komponent korzysta z dwóch stanów: 
1. Własny stan 
2. Stan aplikacji

Może wydać się trochę dziwne to, że `globalState` jest updatowane dopiero po tym, jak był zupdatowany `scheduleState`,(stan komponentu),  zamiast robić to w jednym miejscu, ale to jest związane z tym jak działa react i redux:


1. W momencie kiedy zmieniamy lokalny stan, react zaczyna renderować komponent
2. W momencie zmianu globalnego stanu, wszystkie komponenty są o tym notyfikowane prawie natychmiast
3. W aplikacji jest kilka komponentów lokalny stan których zależy od globalnego
4. z 1. -> przy zmianie globalnego stanu, te komponenty muszą być werenderowane ponownie
5. React na to nie pozwala i wyrzuca bład `Cannot update component (...) during rendering of component (....)`

Rozwiązaniem jest zmieniać globalny stan dopiero kiedy komponent który go zmiania będzie wyrenderowany. Możemy tego osiągnąć za pomocą `useEffect`, który będzie śledzić lokalny stan

Reply:
https://github.com/facebook/react/issues/18178

Reply:
Dobrze wiedzieć. Super, że się tym podzieliłeś :)